A24-lb-policy-config.md

```diff
@@ -0,0 +1,218 @@
+  LoadBalancingConfig fallback_policy = 3;
+}
+
+// Configuration for grpclb LB policy.
+message GrpcLbConfig {
+}
+
+// Selects LB policy and provides corresponding configuration.
+message LoadBalancingConfig {
+  // Exactly one LB policy may be configured.
+  // If no LB policy is configured, then the default behavior is to pick
+  // the first available backend.
+  oneof policy {
+    // For each new LB policy supported by gRPC, a new field must be added
+    // here.  The field's name must be the LB policy name and its type is a
+    // message that provides whatever configuration parameters are needed
+    // by the LB policy.  The configuration message will be passed to the
+    // LB policy when it is instantiated on the client.
+    //
+    // If the LB policy does not require any configuration parameters, the
+    // message for that LB policy may be empty.
+    //
+    // Note that if an LB policy contains another nested LB policy
+    // (e.g., a top-level policy picks the locality and then delegates to
+    // another policy to pick the endpoint within that locality), its
+    // configuration message may include a nested instance of the
+    // LoadBalancingConfig message to configure the nested LB policy.
+    RoundRobinConfig round_robin = 1 [json_name = "round_robin"];
+    XdsConfig xds = 2;
+    GrpcLbConfig grpclb = 3;
+  }
+}
+
+message ServiceConfig {
+  // Load balancing policy.
+  //
+  // Note that load_balancing_policy is deprecated in favor of
+  // load_balancing_config; the former will be used only if the latter
+  // is unset.
+  //
+  // If no LB policy is configured here, then the default behavior is to
+  // pick the first available backend.
+  //
+  // If the deprecated load_balancing_policy field is used, note that if the
+  // resolver returns at least one balancer address (as opposed to backend
+  // addresses), gRPC will use grpclb (see
+  // https://github.com/grpc/grpc/blob/master/doc/load-balancing.md),
+  // regardless of what policy is configured here.  However, if the resolver
+  // returns at least one backend address in addition to the balancer
+  // address(es), the client may fall back to the requested policy if it
+  // is unable to reach any of the grpclb load balancers.
+  enum LoadBalancingPolicy {
+    UNSPECIFIED = 0;
+    ROUND_ROBIN = 1;
+  }
+  LoadBalancingPolicy load_balancing_policy = 1 [deprecated=true];
+  // Multiple LB policies can be specified; clients will iterate through
+  // the list in order and stop at the first policy that they support.
+  // This allows the service config to specify custom policies that may not
+  // be known to all clients.
+  repeated LoadBalancingConfig load_balancing_config = 4;
+
+  // ...other existing fields...
+}
+```
+
+For example, to use the `xds` policy with a child policy of `round_robin`,
+something like this could be used:
+
+```proto
+load_balacing_config: {
+  xds: {
+    balancer_name: "balancer.example.com:8080"
```

Comment:
Should this always include the scheme, so we don't depend the default resolver picked by the user?

Reply:
I think it's probably a good idea to recommend that, and I've updated the examples in this doc.  However, I don't think we necessarily need to enforce this; I think there may be environments where it may be useful to have different clients default to different resolvers that all have a way of resolving the same name.

We can discuss this in more detail when we put together the subsequent gRFC for the xds policy.